samples/net/dns_resolve/src/main.c

```diff
@@ -192,17 +203,24 @@ static void ipv4_addr_add_handler(struct net_mgmt_event_callback *cb,
 	 * management event thread stack is very small by default.
 	 * So run it from work queue instead.
 	 */
-	k_work_init_delayable(&ipv4_timer, do_ipv4_lookup);
-	k_work_reschedule(&ipv4_timer, K_NO_WAIT);
-
-#if defined(CONFIG_MDNS_RESOLVER)
-	k_work_init_delayable(&mdns_ipv4_timer, do_mdns_ipv4_lookup);
-	k_work_reschedule(&mdns_ipv4_timer, K_NO_WAIT);
-#endif
+	schedule_ipv4_queries();
 }
 
 static void setup_dhcpv4(struct net_if *iface)
 {
+	for (int i = 0; i < NET_IF_MAX_IPV4_ADDR; i++) {
+		struct net_if_addr *if_addr =
+			&iface->config.ip.ipv4->unicast[i];
+
+		if (if_addr->addr_type != NET_ADDR_DHCP || !if_addr->is_used) {
+			continue;
+		}
+
+		/* Already have DHCP assigned address, schedule queries. */
+		schedule_ipv4_queries();
+		return;
+	}
+
```

Comment:
I am just wondering that should we have an API function (similar to `net_if_foreach()` so that user supplied callback could be called for each address) that could provide this info. Now we are accessing net_if internals from sample which we should try to avoid.

Reply:
That could be useful, I'll try to introduce such function instead.

Reply:
@jukkar I've added helper functions to loop over IP addresses configured on an interface and made use of it in the sample.